tests/robustness/engine/engine_test.go

```diff
@@ -3,715 +3,728 @@
-	s3RepoBaseDirPath  = filepath.Join(s3BasePath, repoBaseDirName)
-	fsMetadataRepoPath = filepath.Join(fsRepoBaseDirPath, metadataRepoPath)
-	s3MetadataRepoPath = filepath.Join(s3RepoBaseDirPath, metadataRepoPath)
-	fsDataRepoPath     = filepath.Join(fsRepoBaseDirPath, dataRepoPath)
-	s3DataRepoPath     = filepath.Join(s3RepoBaseDirPath, dataRepoPath)
-)
-
-func TestEngineWritefilesBasicFS(t *testing.T) {
-	eng, err := NewEngine("")
-	if err == kopiarunner.ErrExeVariableNotSet || errors.Is(err, fio.ErrEnvNotSet) {
-		t.Skip(err)
-	}
-
-	testenv.AssertNoError(t, err)
-
-	defer func() {
-		cleanupErr := eng.Cleanup()
-		testenv.AssertNoError(t, cleanupErr)
-
-		os.RemoveAll(fsRepoBaseDirPath) //nolint:errcheck
-	}()
-
-	ctx := context.TODO()
-	err = eng.InitFilesystem(ctx, fsDataRepoPath, fsMetadataRepoPath)
-	testenv.AssertNoError(t, err)
-
-	fileSize := int64(256 * 1024)
-	numFiles := 10
-
-	fioOpts := fio.Options{}.WithFileSize(fileSize).WithNumFiles(numFiles)
-
-	err = eng.FileWriter.WriteFiles("", fioOpts)
-	testenv.AssertNoError(t, err)
-
-	snapIDs := eng.Checker.GetSnapIDs()
-
-	snapID, err := eng.Checker.TakeSnapshot(ctx, eng.FileWriter.LocalDataDir)
-	testenv.AssertNoError(t, err)
-
-	err = eng.Checker.RestoreSnapshot(ctx, snapID, os.Stdout)
-	testenv.AssertNoError(t, err)
-
-	for _, sID := range snapIDs {
-		err = eng.Checker.RestoreSnapshot(ctx, sID, os.Stdout)
-		testenv.AssertNoError(t, err)
-	}
-}
-
-func randomString(n int) string {
-	b := make([]byte, n)
-	io.ReadFull(rand.Reader, b) //nolint:errcheck
-
-	return hex.EncodeToString(b)
-}
-
-func makeTempS3Bucket(t *testing.T) (bucketName string, cleanupCB func()) {
-	endpoint := "s3.amazonaws.com"
-	accessKeyID := os.Getenv("AWS_ACCESS_KEY_ID")
-	secretAccessKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
-	sessionToken := os.Getenv("AWS_SESSION_TOKEN")
-	secure := true
-	region := ""
-	cli, err := minio.NewWithCredentials(endpoint, credentials.NewStaticV4(accessKeyID, secretAccessKey, sessionToken), secure, region)
-	testenv.AssertNoError(t, err)
-
-	bucketName = fmt.Sprintf("engine-unit-tests-%s", randomString(4))
-	err = cli.MakeBucket(bucketName, "")
-	testenv.AssertNoError(t, err)
-
-	return bucketName, func() {
-		objNameCh := make(chan string)
-		errCh := cli.RemoveObjects(bucketName, objNameCh)
-		go func() {
-			for removeErr := range errCh {
-				t.Errorf("error removing key %s from bucket: %s", removeErr.ObjectName, removeErr.Err)
-			}
-		}()
-
-		recursive := true
-		doneCh := make(chan struct{})
-		defer close(doneCh)
-
-		for obj := range cli.ListObjects(bucketName, "", recursive, doneCh) {
-			objNameCh <- obj.Key
-		}
-
-		close(objNameCh)
-
-		retries := 10
-		retryPeriod := 1 * time.Second
-		var err error
-
-		for retry := 0; retry < retries; retry++ {
-			time.Sleep(retryPeriod)
-			err = cli.RemoveBucket(bucketName)
-			if err == nil {
-				break
-			}
-		}
-		testenv.AssertNoError(t, err)
-	}
-}
+// import (
```

Comment:
Please ignore this. Had to comment it out as the methods tested here are also commented out.